internal/database/database.go

```diff
@@ -0,0 +1,485 @@
+// Package database provides persistent storage for auth tokens, jobs, and flavors,
+// and computes "pressure" for each flavor.
+//
+// Pressure is a numeric measure of demand for runner machines (flavors). It is
+// defined as the number of incomplete jobs (completed_at IS NULL) assigned to a
+// given flavor.
+//
+// When a job is inserted, it is assigned a flavor using the following algorithm:
+//  1. The flavor's platform must match the job's platform.
+//  2. The flavor's labels must be a superset of (or equal to) the job's labels.
+//  3. If multiple flavors match, choose the one with the highest priority.
+//  4. If there is still a tie, pick one at random.
```

Comment:
Can we put this comment directly above line 29 ? I fear it will otherwise not be kept in sync with changes to the algorithm.

Reply:
This is the equivalent of Python docstrings, called [go doc](https://tip.golang.org/doc/comment) for Go packages. It needs to be at the beginning of a package.